### PR TITLE
rustfmt tests

### PR DESCRIPTION
**Pull request by matthiaskrgr**
also fails CI if tests turn out to not be formatted properly.

### CODE REVIEW

File: tests/ui/attrs.rs [hunk 1]
```diff
@@ -30,22 +26,27 @@ fn false_positive_stmt() {
 
 #[inline(always)]
 fn empty_and_false_positive_stmt() {
-    ;
```
Comment (sinkuu) [inline tests/ui/attrs.rs]: This function needs `#[rustfmt::skip]`

File: tests/ui/decimal_literal_representation.rs [hunk 2]
```diff
@@ -7,24 +7,22 @@
...
-        127,            // 0x7F
-        256,            // 0x100
-        511,            // 0x1FF
-        2048,           // 0x800
-        4090,           // 0xFFA
-        16_371,         // 0x3FF3
-        61_683,         // 0xF0F3
-        2_131_750_925,  // 0x7F0F_F00D
+    let good = (
+        // Hex:
```
Comment (phansch) [inline tests/ui/decimal_literal_representation.rs]: nit: I think this was supposed to be some in a column with the values below

File: tests/ui/decimal_literal_representation.rs [hunk 3]
```diff
@@ -7,24 +7,22 @@
...
+        511,           // 0x1FF
+        2048,          // 0x800
+        4090,          // 0xFFA
+        16_371,        // 0x3FF3
+        61_683,        // 0xF0F3
+        2_131_750_925, // 0x7F0F_F00D
     );
-    let bad = (        // Hex:
+    let bad = (
+        // Hex:
```
Comment (phansch) [inline tests/ui/decimal_literal_representation.rs]: nit: same for this comment

File: tests/ui/implicit_hasher.rs [hunk 4]
```diff
@@ -85,19 +82,19 @@ macro_rules! gen {
...
     (fn $name:ident) => {
-        pub fn $name(_map: &mut HashMap<i32, i32>, _set: &mut HashSet<i32>) {
-        }
-    }
+        pub fn $name(_map: &mut HashMap<i32, i32>, _set: &mut HashSet<i32>) {}
+    };
 }
 
-gen!(impl);
+gen!(impl );
```
Comment (phansch) [inline tests/ui/implicit_hasher.rs]: This looks like a bug in rustfmt? Might be good to make a small example out of it and report it.

File: tests/ui/swap.rs [hunk 5]
```diff
@@ -54,7 +50,7 @@ fn main() {
     a = b;
     b = a;
 
-    ; let t = a;
+;    let t = a;
```
Comment (phansch) [inline tests/ui/swap.rs]: I'm not sure if this change is desired. Is it a bug in rustfmt?

Comment (matthiaskrgr) [reply tests/ui/swap.rs]: Probably, I already filed a ticket for this (https://github.com/rust-lang/rustfmt/issues/3245), can `[rustfmt::skip]` later.

File: ci/base-tests.sh [hunk 6]
```diff
@@ -29,3 +29,21 @@ cd clippy_dev && cargo test && cd ..
...
 cargo +nightly fmt --all -- --check
+
+
+#avoid loop spam
+set +ex
+# make sure tests are formatted
+
+# some lints are sensitive to formatting, exclude some files
+needs_formatting=false
+for file in `find tests -not -path "tests/ui/methods.rs" -not -path "tests/ui/format.rs" -not -path "tests/ui/formatting.rs" -not -path "tests/ui/empty_line_after_outer_attribute.rs" -not -path "tests/ui/double_parens.rs" -not -path "tests/ui/doc.rs" -not -path "tests/ui/unused_unit.rs" | grep "\.rs$"` ; do
```
[the remark this answers is not in the document]
Comment (matthiaskrgr) [reply ci/base-tests.sh]: It does, `find` recurses into subdirs by default.
It looks like `author/` is already formatted :)

[the remark this answers is not in the document]
Comment (phansch) [reply ci/base-tests.sh]: oh, alright then!

File: tests/ui/implicit_hasher.rs (hunk 4 above)
Comment (matthiaskrgr) [reply tests/ui/implicit_hasher.rs]: ticket: https://github.com/rust-lang/rustfmt/issues/3051

**Comment by oli-obk** [comment]
appveyor is unhappy

**Comment by phansch** [comment]
@oli-obk yup, that's probably going to be fixed by https://github.com/rust-lang-nursery/failure/pull/279

**Comment by matthiaskrgr** [comment]
./ci/base-tests.sh: line 44: [: missing `]`
I should check that out

**Comment by matthiaskrgr** [comment]
The fact that CI was green while there was a syntax error in the script is actually quite concerning...
I'll look into this.

**Comment by mati865** [comment]
I'd didn't exit because it was told not to exit on error https://github.com/rust-lang/rust-clippy/blob/921d4da103ad1248508f3cc1b4c642b0b5b1b6d2/ci/base-tests.sh#L35

**Comment by matthiaskrgr** [comment]
Fixed via #3537

**Pull request merged**